allensdk/brain_observatory/behavior/running_processing.py

```diff
@@ -14,40 +16,378 @@ def calc_deriv(x, time):
     return dxdt
 
 
-def deg_to_dist(speed_deg_per_s):
-    '''
-    takes speed in degrees per second
-    converts to radians
-    multiplies by radius (in cm) to get linear speed in cm/s
-    '''
+def _angular_change(summed_voltage: np.ndarray,
+                    vmax: Union[np.ndarray, float]) -> np.ndarray:
+    """
+    Compute the change in degrees in radians at each point from the
+    summed voltage encoder data.
+
+    Parameters
+    ----------
+    summed_voltage: 1d np.ndarray
+        The "unwrapped" voltage signal from the encoder, cumulatively
+        summed. See `_unwrap_voltage_signal`.
+    vmax: 1d np.ndarray or float
+        Either a constant float, or a 1d array (typically constant)
+        of values. These values represent the theoretical max voltage
+        value of the encoder. If an array, needs to be the same length
+        as the summed_voltage array.
+    Returns
+    -------
+    np.ndarray
+        1d array of change in degrees in radians from each point
+    """
+    delta_theta = np.diff(summed_voltage, prepend=np.nan) / vmax * 2 * np.pi
+    return delta_theta
+
+
+def _shift(
+        arr: Iterable,
+        periods: int = 1,
+        fill_value: float = np.nan) -> np.ndarray:
+    """
+    Shift index of an iterable (array-like) by desired number of
+    periods with an optional fill value (default = NaN).
+
+    Parameters
+    ----------
+    arr: Iterable (array-like)
+        Iterable containing numeric data. If int, will be converted to
+        float in returned object.
+    Returns
+    -------
+    np.ndarray (1d)
+        Copy of input object as a 1d array, shifted.
+    """
+    if fill_value is None:
+        fill_value = np.nan
+    if isinstance(fill_value, float):
+        # Circumvent issue if int-like array with np.nan as fill
+        shifted = np.roll(arr, periods).astype(float)
+    else:
+        shifted = np.roll(arr, periods)
+    shifted[:periods] = fill_value
+    return shifted
+
+
+def deg_to_dist(angular_speed: np.ndarray) -> np.ndarray:
+    """
+    Takes the angular speed (radians/s) at each step in radians, and
+    computes the linear speed in cm/s.
+
+    Parameters
+    ----------
+    angular_speed: np.ndarray (1d)
+        1d array of angular speed in radians/s
+    Returns
+    -------
+    np.ndarray (1d)
+        Linear speed in cm/s at each time point.
+    """
     wheel_diameter = 6.5 * 2.54  # 6.5" wheel diameter
     running_radius = 0.5 * (
-        2.0 * wheel_diameter / 3.0)  # assume the animal runs at 2/3 the distance from the wheel center
-    running_speed_cm_per_sec = np.pi * speed_deg_per_s * running_radius / 180.
+        # assume the animal runs at 2/3 the distance from the wheel center
+        2.0 * wheel_diameter / 3.0)
```

Comment:
something looks off here. The speed should be calculated in terms of circumference, not diameter. Why is `pi` removed?

Reply:
The units are staying in radians rather than being converted to degrees elsewhere.

Reply:
No, wheel diameter is definitely the wheel diameter... I would talk over this with @dougollerenshaw if you have concerns, because this is the original implementation and reproduces the results with the expected magnitude.

Reply:
I see, there are 2*pi cancelling in the next line